aggregator_core/src/datastore.rs

```diff
@@ -5905,6 +6041,42 @@ pub mod models {
 
         to_sql_checked!();
     }
+
+    #[derive(Debug, Clone, PartialEq, Eq)]
+    pub struct GlobalHpkeKeypair {
+        hpke_keypair: HpkeKeypair,
+        is_active: bool,
+        expired_at: Option<Time>,
```

Comment:
Q: why time-based expiry, instead of an enum with values like `Active`, `Primary`, `Expired`?

Reply:
We have to keep track of the expiry time either way to determine when the key can be fully dropped from the database. It's a bit of a tradeoff--the state is implicit instead of explicit, but in exchange we don't have to maintain coherency between an `expired_at` column and state enum.

It's fair enough to record `created_at` and an explicit state though. That actually might work a lot nicer when it comes to determining advertise-able keypairs.

Reply:
Explicit state works out a lot nicer, simplifies the datastore API and tests. Great suggestion!